### DISCUSSION TITLE
Possibly least elegant find/replace/cut?

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1094] [Show and tell]
Pulling gene symbol out of fasta title line...

```
head -1 ../analyses/Caenorhabditis_elegans.WBcel235.pep.all.tab
Y110A7A.10.1	pep chromosome:WBcel235:I:5107833:5110183:1 gene:WBGene00000001 transcript:Y110A7A.10.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:aap-1 description:Phosphoinositide 3-kinase adapter subunit; Phosphoinositide kinase AdAPter subunit  [Source:UniProtKB/TrEMBL;Acc:G5EDP9]	MSTTPGTPHGVTHSLMEQGWYWADADRSAVSKALSDQPDGSFIVRNASTPGDYTLSVKFAAQVKLLRIVVKDGKCGFNTDSLTHDSVVRLIEFHRNISLNIFNDALDVRLLYPVSVRRNSQNGKPFFKKGQLQQRMILSAKNDHDWRERLEMENLRAVHLAFERGAKLYDSAHQEMERAESLYHALNQSVRDNEIKLGKLNNLLEAETDVVTQIQGSQTTSEMLKGAFANNKTFVEESIRRINTELTSSKDKKKTLSGILDEIAMKRSNSKTRLCKLMELRSAVYDKMEPALCSRMAAMLDAGAEMINSEPTKVTQLLVDMELKWTPAQYLMCGTSKENAANALIHARYRIAQLDKAVGLKREPQDGIFLIRGSSSHGDKLVLSVLHGERVSHCLIEQNEEGWGFEHSNVYLTTISDFVRYYSHFSLETHADAIKTPLRTPAFDAVTSDTSKPLRNGPGQVFTALPISRKYMEKALLTDDPTKPPESPEPRPDSWDSPSPDTPSSSDSRLVHPTTLPKSIPE
```

```
head -1 ../analyses/Caenorhabditis_elegans.WBcel235.pep.all.tab | sed 's/gene_symbol:/@/g' |   sed 's/ description:/@ /g' | tr '@' "\t" | cut -f1,3
```

``` 
Y110A7A.10.1	aap-1
```

**Comment by kubu4** [comment]
This is a bit shorter:

`head -1 ../analyses/Caenorhabditis_elegans.WBcel235.pep.all.tab | awk -F "[: \t]" '{print $1 "\t" $18}'` 

EDITED: Fixed copy/paste mistake.

**Reply by sr320** [reply]
This assumes there is a consistent number of `:` prior to  gene_symbol? basically, setting that as a delimiter?

**Reply by kubu4** [reply]
Although, my example is probably less robust, in that it expects the number of fields to be consistent throughout...

This type of parsing is probably better suited to a higher level language where you can define a list of categories and then pull out values for each specific category.

**Reply by kubu4** [reply]
> This assumes there is a consistent number of `:`

Yep, my example sets three delimiters: colon, space, and tabs.

**Comment by kubu4** [comment]
This seems like a solid solution, and is very clever, although it seems like you're not using this part: 

`sed 's/ description:/@ /g'`
